validator/js/engine/parse-srcset.js

```diff
@@ -84,7 +84,7 @@ const parseSrcset = function(srcset) {
           '(?:,\\s*)?' +
           '([^,\\s]\\S*[^,\\s])' +
           '\\s*' +
-          '([\\d]+.?[\\d]*[w|x])?' +
+          '([\\d]+(?:\\.[\\d]+)?[w|x])?' +
           '\\s*' +
           '(?:(,)\\s*)?',
```

Comment:
One other thing may be missing here:

According to the [spec definition](https://html.spec.whatwg.org/multipage/images.html#width-descriptor) of a `srcset attribute`, it states that:

- a width descriptor is a number greater than zero representing the width descriptor value, and a U+0077 LATIN SMALL LETTER W character
- a pixel density descriptor is a valid floating-point number giving a number greater than zero representing the pixel density descriptor value, and a U+0078 LATIN SMALL LETTER X character

So that would mean float numbers should only match with the `x` character, and integers can either be matched with the `w` or `x` character. Currently, that's not the case.

Reply:
I've adapted the regex to be more direct representation of the above definition, including the requirement that the values should be greater than zero.

Reply:
Resolved via https://github.com/ampproject/amphtml/pull/30252/commits/d69e66dc1cb451c593ac832d96b3f1d9d8140607